src/views/bridge-confirmation/bridge-confirmation.view.tsx

```diff
@@ -188,21 +133,29 @@ const BridgeConfirmation: FC = () => {
               });
             }
           });
-      }
-    };
-    void estimateGas();
-    estimateGasPollInterval.current = window.setInterval(estimateGas, AUTO_REFRESH_RATE);
-  }, [connectedProvider, formData, tokenBalance, callIfMounted, estimateBridgeGas, notifyError]);
+      };
 
-  const stopGasPolling = useCallback(() => {
-    clearInterval(estimateGasPollInterval.current);
-  }, []);
+      connectedProvider.data.provider.on("block", onBlock);
```

Comment:
I think that we need to apply a debouncer for this event. According to my tests, it can be fired several times at once for accumulated blocks.

Reply:
The thing is that we don't need to call the `onBlock` for the first time, as it already fires with the last mined block when you attach a listener to it. Reverted!

Reply:
Not sure why do you mean by "accumulated blocks", it gets fired for the current block + each new mined block in the chain. Of course time between blocks may vary. Not sure if the debouncing would ver very helpful here as the time between blocks is random, so probably we'll debounce just one request more times than two sequential requests...

Reply:
> Not sure why do you mean by "accumulated blocks"

I mean that the callback might not be called in 20 seconds and the called 3 times in half a second.

> The best option would be to abort the previous request when a new block is available, but it's not possible with `ethers`.

That's why I propose a debouncer.

Reply:
That's not happening to me, how did you manage to get all that blocks accumulated? Blocks in Ethereum happens each 10s more or less (in Goerli the block time is even higher), so it's impossible to get 3 of them together.

Reply:
It looks like Ethers [queries from their last queried block to the current block](https://github.com/ethers-io/ethers.js/issues/492#issuecomment-491519871) and sometimes the event is fired several times in a row.

We have implemented a debouncer to mitigate this issue.